_posts/2016-03-18-functional-architecture-is-ports-and-adapters.html

```diff
@@ -707,7 +707,7 @@ <h2 id="comments-header">Comments</h2>
 				In Haskell, you don't have to read the source code in order to know whether a function is impure or not. You can simply look at the type!
 			</p>
 			<p>
-				This is one of the many benefits of pure functions: we know, by definition, that they have no side-effects. This property holds not only for the function itself, but for all functions that that pure function calls. This follows from the definition of purity. A pure function can call other pure functions, but it can't call impure functions, because, if it did, it would become impure itself. Haskell enforces this rule, so when you're looking at a pure function, you know that that entire subgraph of your code is pure. Otherwise, it wouldn't compile.
+				This is one of the many benefits of pure functions: we know, by definition, that they have no side-effects. This property holds not only for the function itself, but for all functions that a pure function calls. This follows from the definition of purity. A pure function can call other pure functions, but it can't call impure functions, because, if it did, it would become impure itself. Haskell enforces this rule, so when you're looking at a pure function, you know that that entire subgraph of your code is pure. Otherwise, it wouldn't compile.
```

Comment:
This change look wrong to me. The second _that_ in _that that_ refers back to _the function itself_, whereas the first _that_ is interchangeable with _which_.